### PR TITLE
Stage 1 of clang complier warning patches. These are mostly fixes for…

### PR DESCRIPTION
**Pull request by ethoms**
… data type and casting inconsistencies. Also some string formatting fixes.

These are patches that come from a code cleanup exercise trying to clear as many clang compiler warnings as possible. But they should apply to any build environement.
 
Note that the SOPE and SOGo both build sucessfully after these and many other patches have been applied. Also, I've been running in production for a few days without any issues. However I have not done any unit testing.

### CODE REVIEW

File: sope-core/NGStreams/NGByteCountStream.m [hunk 1]
```diff
@@ -87,7 +87,7 @@ - (unsigned)readBytes:(void *)_buf count:(unsigned)_len {
   {
     register unsigned char *byteBuffer = _buf;
 
-    for (_len = result - 1; _len >= 0; _len--, byteBuffer++) {
+    for (NSInteger len = result - 1; len >= 0; len--, byteBuffer++) {
```
Comment (buzzdeee) [inline sope-core/NGStreams/NGByteCountStream.m]: This change broke my build on macppc, libobjc2, gcc-4.2.1:
NGByteCountStream.m: In function '-[NGByteCountStream readBytes:count:]':
NGByteCountStream.m:90: error: 'for' loop initial declaration used outside C99 mode

File: sope-core/NGStreams/NGByteCountStream.m [hunk 2]
```diff
@@ -106,7 +106,7 @@ - (unsigned)writeBytes:(const void *)_buf count:(unsigned)_len {
   {
     register unsigned char *byteBuffer = (unsigned char *)_buf;
 
-    for (_len = result - 1; _len >= 0; _len--, byteBuffer++) {
+    for (NSInteger len = result - 1; len >= 0; len--, byteBuffer++) {
```
Comment (buzzdeee) [inline sope-core/NGStreams/NGByteCountStream.m]: same here as above:

NGByteCountStream.m: In function '-[NGByteCountStream writeBytes:count:]':
NGByteCountStream.m:109: error: 'for' loop initial declaration used outside C99 mode

I'll provide separate PR that fixes the issue.

File: sope-core/NGStreams/NGByteCountStream.m (hunk 1 above)
Comment (buzzdeee) [inline sope-core/NGStreams/NGByteCountStream.m]: and I see that is already addressed in 37782867bc35981da0ed8593f9389e9761c3f83a, same thing I came up with by myself ;)

**Comment by ethoms** [comment]
These have already been fixed by extrafu! What branch are you using? You should have the commit "(fix) c99 fixes" (37782867bc35981da0ed8593f9389e9761c3f83a).

**Pull request merged**